syslog/syslog.go

```diff
@@ -90,25 +90,20 @@ type Logger struct {
 
 // Dial connects to the syslog server at raddr, using the optional certBundle,
 // and launches a goroutine to watch logger.Packets for messages to log.
-func Dial(clientHostname, network, raddr string, rootCAs *x509.CertPool) (*Logger, error) {
-	// dial once, just to make sure the network is working
-	conn, err := dial(network, raddr, rootCAs)
-
-	if err != nil {
-		return nil, err
-	} else {
-		logger := &Logger{
-			ClientHostname: clientHostname,
-			network:        network,
-			raddr:          raddr,
-			rootCAs:        rootCAs,
-			Packets:        make(chan Packet, 100),
-			Errors:         make(chan error, 0),
-			conn:           conn,
-		}
-		go logger.writeLoop()
-		return logger, nil
+func Dial(clientHostname, network, raddr string, rootCAs *x509.CertPool) (*Logger) {
+	logger := &Logger{
+		ClientHostname: clientHostname,
+		network:        network,
+		raddr:          raddr,
+		rootCAs:        rootCAs,
+		Packets:        make(chan Packet, 100),
+		Errors:         make(chan error, 0),
 	}
+
+	logger.connect()
```

Comment:
Ah, yes, I hadn't thought of that. The problem is if I go back to a single `dial()` and return it, there's nothing to make it retry and we'd have to wait until a log message is sent to see any other errors.

I wonder if an `init` method could be added to create an Error channel before `Dial()` is called in `remote_syslog.go`, and then a goroutine could run:

```
for err := range logger.Errors {
 		log.Errorf("Syslog error: %v", err)
 	}
```

What do you think?